.github/workflows/build-upload-deploy-prod.yml

```diff
@@ -0,0 +1,116 @@
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build App
+        env:
+          NODE_ENV: "production"
+          API_ENDPOINT: "https://api.seize.io"
+          ALLOWLIST_API_ENDPOINT: "https://allowlist-api.seize.io"
+          BASE_ENDPOINT: "https://seize.io"
+          LOAD_S3: "true"
+          VERSION: ${{ env.COMMIT_SHA }}
+          ALCHEMY_API_KEY: ${{ secrets.ALCHEMY_API_KEY }}
+        run: npm run build
+
+      - name: Build Target
+        run: |
+          mkdir -p target/_next
+          cp -r .next/static target/_next/
+          zip -r package.zip . -x '*.git*' 'node_modules/*' 'target/*'
+          mv package.zip target/
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Upload assets to S3
+        run: aws s3 sync target s3://${{ env.AWS_BUCKET }}/web_build/${{ env.COMMIT_SHA }}/
+
+      - name: Get commit message
+        run: echo "COMMIT_MESSAGE=$(git show -s --format=%s)" >> $GITHUB_ENV
+
+      - name: Create new ElasticBeanstalk Application Version
+        run: |
+          aws elasticbeanstalk create-application-version \
+          --application-name "${{ env.BEANSTALK_APP_NAME }}" \
+          --source-bundle S3Bucket="${{ env.AWS_BUCKET }}",S3Key="web_build/${{ env.COMMIT_SHA }}/package.zip" \
+          --version-label "${{ env.COMMIT_SHA }}" \
+          --description "${{ env.COMMIT_MESSAGE }}"
+
+      - name: Deploy new Version to ElasticBeanstalk
+        run: |
+          aws elasticbeanstalk update-environment \
+          --application-name "${{ env.BEANSTALK_APP_NAME }}" \
+          --environment-name "${{ env.BEANSTALK_ENV_NAME }}" \
+          --version-label "${{ env.COMMIT_SHA }}"
+
+      - name: Check Elastic Beanstalk health and readiness (60s warmup then 30 retries with 15s delay)
+        run: |
+          echo "Warmup..."
+          MAX_RETRIES=60
+          COUNTER=0
+          sleep 30 # Initial sleep for the new EC2 to initialize
+          while : ; do
+            ENV_HEALTH=$(aws elasticbeanstalk describe-environments --environment-names ${{ env.BEANSTALK_ENV_NAME }} --query 'Environments[0].Health' --output text)
+            ENV_STATUS=$(aws elasticbeanstalk describe-environments --environment-names ${{ env.BEANSTALK_ENV_NAME }} --query 'Environments[0].Status' --output text)
+            echo "Environment health: $ENV_HEALTH"
+            echo "Environment status: $ENV_STATUS"
+            if [[ $ENV_HEALTH == "Green" && $ENV_STATUS == "Ready" ]]; then
+              echo "Environment is healthy and ready!"
+              break
+            elif [[ $COUNTER -eq $MAX_RETRIES ]]; then
```

Comment:
what if during those 30 tries aws does: deploy -> pending -> failed -> rollback -> success
wouldn't we think then that all's good but it really isn't?

Reply:
i believe you would see the logs in git workflow but any suggestions on this?

Reply:
As a countermeasure we could ask the currently deployed version from eb and compare (as the last step)